Source/DafnyCore/Resolver.cs

```diff
@@ -15725,6 +15725,14 @@ public ResolutionContext(ICodeContext codeContext, bool isTwoState)
       var types = new Type[] { type0, type1 };
       AllXConstraints.Add(new XConstraint(tok, constraintName, types, errMsg));
     }
+    private void AddXConstraint(IToken tok, string constraintName, Type[] types, string errMsgFormat) {
```

Comment:
Are you using this new function?

Reply:
Oops, good catch - I ended up not needing this. I've removed it now.